framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3506,7 +3506,11 @@ public boolean validateType(Tree tree, AnnotatedTypeMirror type) {
         // basic consistency checks
         if (!AnnotatedTypes.isValidType(atypeFactory.getQualifierHierarchy(), type)) {
             checker.report(
-                    Result.failure("type.invalid", type.getAnnotations(), type.toString()), tree);
```

Comment:
Is this changed error key really an improvement? This error gets raised, e.g. when too few annotations are present. Is the "conflicting" message really clearer than the "type.invalid" message?
How about keeping `type.invalid` here and adding an issue to give a more precise error message, like 'type.invalid.too.few.annotations`, 'type.invalid.too.many.annotations`, etc.
The logic from `checkConflictingPrimaryAnnos` could then be used to raise `type.invalid.conflicting.annos`.
(`too.many` is probably the same as `conflicting`?)